infra/modules/auth-github-actions/main.tf

```diff
@@ -31,7 +31,8 @@ data "tls_certificate" "github" {
 }
 
 locals {
-  oidc_thumbprint_github = data.tls_certificate.github.certificates.0.sha1_fingerprint
+  github_certificates    = data.tls_certificate.github.certificates
+  oidc_thumbprint_github = local.github_certificates[length(local.github_certificates) - 1].sha1_fingerprint
```

Comment:
We ended up doing a hacky way to fix it since we got a third thumbprint instead of the two that was in the GitHub page. BTW, using the splat (`*`) lets you pull more than just the first one

```
locals {
  # This is used when the thumbprints from GitHub are not every one that they
  # say to include. The response from the data is not consistent, and changes often
  # after being ran sequentially. This is a stopgap while we figure out a better solution
  temp_other_github_thumbprints = [
    "1c58a3a8518e8759bf075b76b750d4f2df264fcd",
    "6938fd4d98bab03faadb97b34396831e3780aea1"
  ]
}

data "tls_certificate" "github" {
  url = "https://token.actions.githubusercontent.com"
}
```

And use it like this so we remove the duplicates

```
  thumbprint_list = distinct(concat(
    data.tls_certificate.github.certificates[*].sha1_fingerprint,
    local.temp_other_github_thumbprints
))
```

Reply:
my understanding based on the AWS docs is that we just need the last certificate sha1. why do we need all of the intermediate certs?

also, do we need the `temp_other_github_thumbprints`?

Reply:
I did more than one because every time I ran our code, it would give me one of the temp ones, which is the two that GitHub says to have, or a third one. Since it was always random, I opened it up to any thumbprint it responded with, which was one from the temp list and the third one. I used the temp list to make sure I covered both of the ones listed, and if there were new ones, those

Reply:
Our original thumbprint was `6938fd4d98bab03faadb97b34396831e3780aea1`, and when I ran the code again to see if we got a new one, we'd get a mix of one of these three

```
"6938fd4d98bab03faadb97b34396831e3780aea1",
"f879abce0008e4eb126e0097e46620f5aaae26ad",
"1c58a3a8518e8759bf075b76b750d4f2df264fcd"
```

Opening up the thumbprint with the splat would give me 

```
"6938fd4d98bab03faadb97b34396831e3780aea1",
"f879abce0008e4eb126e0097e46620f5aaae26ad"
```

Or

```
"f879abce0008e4eb126e0097e46620f5aaae26ad",
"1c58a3a8518e8759bf075b76b750d4f2df264fcd"
```

but never both listed from [here](https://github.blog/changelog/2023-06-27-github-actions-update-on-oidc-integration-with-aws/), which is why I added the temp list to make sure those were in there at least, but keep it from being just hard-coded

Reply:
Oh I see. Sigh, darnit github why they gotta make this complicated

Reply:
@Nava-JoshLong when I open up the thumbprint with the splat I see:

```
"1c58a3a8518e8759bf075b76b750d4f2df264fcd",
"f879abce0008e4eb126e0097e46620f5aaae26ad"
```

or 

```
"6938fd4d98bab03faadb97b34396831e3780aea1",
"f879abce0008e4eb126e0097e46620f5aaae26ad",
```

<img width="676" alt="image" src="https://github.com/navapbc/template-infra/assets/447859/93cacecc-86eb-44d6-b70e-41c086c20d86">

i`"f879abce0008e4eb126e0097e46620f5aaae26ad"` is always the last one in the chain. the two that github put in the blog post are always the first ones i.e. the intermediate ones.

so i think the fix that i have here should actually work, and is what AWS recommends in their docs.

adding the the earlier steps in the chain is unnecessary i think.

Reply:
Makes sense. We'll keep it in our just for another week or so until we have time for circling back and see if this works with just the splat. The fact it keeps changing between applies makes me want to be cautious and keep the 3 for now and watch what Github does

Reply:
You actually don't want to use the splat, you want just the last certificate. Otherwise terraform will randomly show a diff in the terraform plan vs what's in the tfstate

Reply:
Anyways, @Nava-JoshLong are you comfortable with this fix in this template? If so can you approve?